Sources/Fuzzilli/Corpus/Corpus.swift

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Protocol defining a Corpus for use in Fuzzilli
+/// It manages discovered programs, determines the next seed to target, 
+/// and specifies how much energy to allocate to that seed
+
+public protocol Corpus : Component {
+    var size: Int { get }
+    var isEmpty: Bool { get }
+    // Whether or not the corpus requires edge tracking to function
+    var requiresEdgeTracking: Bool { get }
```

Comment:
Instead of exposing this, could the CoverageEvaluator maybe expose a `enableEdgeCounting` method which the `MarkovCorpus.initialize` would call (probably this would require a downcast, but I think that's fine, if it fails, there would just be a `logger.fatal("The Markov Corpus requires coverage feedback")` or so)? If the edge selection code is moved into swift code (see below), then this method could probably just return the raw Int64 array into which the edge counts will be written (or it could wrap that array into a struct for convenience)